File: src/bisbalang.js
const dataCount = 30000;
let data, pointer;

function run()
{
    data = new Array(dataCount).fill(0);
    pointer = 0;
    document.getElementById("outputTextbox").value = "";

    let inputPointer = 0;

    let program = document.getElementById("inputTextbox").value.split(" ");

    if(program.length % 2 != 0)
    {
        document.getElementById("outputTextbox").value = "Error: Wrong count of words (" + program.length.toString() + ").";
        return;
    }

    for(let i = 0; i < program.length; i+=2)
    {
        let wordA = program[i], wordB = program[i+1];

        if(wordA == "camina" && wordB == "y") pointer = (pointer + 1) % dataCount;
        else if(wordA == "y" && wordB == "camina") pointer = (pointer == 0) ? dataCount - 1 : pointer - 1;
        else if(wordA == "camina" && wordB == "camina") data[pointer] = (data[pointer] + 1) % 256;
        else if(wordA == "ven" && wordB == "ven") data[pointer] = (data[pointer] == 0) ? 255 : data[pointer] - 1;
        else if(wordA == "ven" && wordB == "camina") document.getElementById("outputTextbox").value += String.fromCharCode(data[pointer]);
        else if(wordA == "camina" && wordB == "ven") data[pointer] = document.getElementById("programInputTextbox").value[inputPointer++].charCodeAt(0);
        else if(wordA == "ven" && wordB == "y") if(data[pointer] == 0) while(!(program[i] == "y" && program[i+1] == "ven")) i += 2;
        else if(wordA == "y" && wordB == "ven") if(data[pointer] != 0) while(!(program[i] == "ven" && program[i+1] == "y")) i -= 2;
        else
        {
            document.getElementById("outputTextbox").value = "Error: Unknown instruction at word " + i.toString() + ".";
            return;
        }
    }
}

File: src/bisbalang.cpp
#include <fstream>
#include <iostream>
#include <string>

using namespace std;

#define DATA_SIZE 30000

// camina = Ook.
// y = Ook?
// ven = Ook!

int main()
{
    unsigned char data[DATA_SIZE];
    unsigned int pointer = 0;

    string filename;
    cout << "File name: ";
    cin >> filename;

    ifstream file(filename);
    if(!file.is_open())
    {
        cout << "Couldn't open file " << filename << endl;
        return 1;
    }

    string compiled;
    while(!file.eof())
    {
        string wordA, wordB;
        file >> wordA >> wordB;

        int instrA, instrB;
        if(wordA == "camina") instrA = 1;
        else if(wordA == "y") instrA = 2;
        else if(wordA == "ven") instrA = 3;
        else
        {
            cout << "Unknown instruction: " << wordA << endl;
            return 1;
        }

        if(wordB == "camina") instrB = 1;
        else if(wordB == "y") instrB = 2;
        else if(wordB == "ven") instrB = 3;
        else
        {
            cout << "Unknown instruction: " << wordB << endl;
            return 1;
        }
        
        if(instrA == 1 && instrB == 2) compiled += '>';
        else if(instrA == 2 && instrB == 1) compiled += '<';
        else if(instrA == 1 && instrB == 1) compiled += '+';
        else if(instrA == 3 && instrB == 3) compiled += '-';
        else if(instrA == 3 && instrB == 1) compiled += '.';
        else if(instrA == 1 && instrB == 3) compiled += ',';
        else if(instrA == 3 && instrB == 2) compiled += '[';
        else if(instrA == 2 && instrB == 3) compiled += ']';
        else
        {
            cout << "Unknown instruction: " << wordA << " " << wordB << endl;
            return 1;
        }
    }

    for(int i = 0; i < compiled.length() && i >= 0; i++)
    {
        switch(compiled[i])
        {
            case '>': pointer++; break;
            case '<': pointer--; break;
            case '+': data[pointer]++; break;
            case '-': data[pointer]--; break;
            case '.': cout << data[pointer]; break;
            case ',': cin >> data[pointer]; break;
            case '[':
            {
                if(data[pointer] == 0) while(compiled[i] != ']' && i < compiled.length()) i++;
                break;
            }
            case ']':
            {
                if(data[pointer] != 0) while(compiled[i] != '[' && i >= 0) i--;
                break;
            }
        }
    }
}